client/src/app/pages/applications/applications-table/applications-table.tsx

```diff
@@ -1302,6 +1359,33 @@ export const ApplicationsTable: React.FC = () => {
           }}
         />
       </div>
+      <Modal
+        isOpen={isDownloadModalOpen}
+        variant="small"
+        title={t("actions.download", { what: "analysis details reports" })}
+        onClose={() => setIsDownloadModalOpen(false)}
+      >
+        <FormGroup label="Select Format" fieldId="format-select">
+          <div>
+            <Button
+              variant={selectedFormat === "json" ? "primary" : "secondary"}
+              onClick={() => setSelectedFormat("json")}
+            >
+              {<CodeIcon />} JSON
+            </Button>
+            <Button
+              variant={selectedFormat === "yaml" ? "primary" : "secondary"}
+              onClick={() => setSelectedFormat("yaml")}
+            >
+              {<CodeIcon />} YAML
+            </Button>
+          </div>
+          <p>Selected Format: {selectedFormat}</p>
+        </FormGroup>
+        <Button variant="primary" onClick={handleDownload}>
+          {t("actions.download")}
+        </Button>
+      </Modal>
```

Comment:
The look of the modal needs to be updated.

Have a look at the "Show dropdown modal": https://www.patternfly.org/components/modal#with-dropdown

Put the modal's action buttons in the `actions` parameter so they render in the correct place with all the good spacing.


Put the yaml vs json selection in a basic drop down: https://www.patternfly.org/components/forms/form-select#basic